src/runtime_src/hip/api/hipMemory.cpp

```diff
@@ -0,0 +1,222 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023 Advanced Micro Device, Inc. All rights reserved.
+
+#include <string>
+#include "hip/config.h"
+#include "hip/hip_runtime_api.h"
+#include "hip/core/device.h"
+#include "hip/core/memory.h"
+#include "core/common/error.h"
+
+namespace xrt::core::hip {
+
+// Allocate memory on the device.
+static void
+hipMalloc(void** ptr, size_t size)
+{
+    throw std::runtime_error("Not implemented");
+}
+
+// Allocates device accessible host memory.
+static void
+hipHostMalloc(void** ptr, size_t size, unsigned int flags)
+{
+    throw std::runtime_error("Not implemented");    
+}
+
+// Free memory allocated by the hipHostMalloc().
+static void 
+hipHostFree(void* ptr)
+{
+    throw std::runtime_error("Not implemented");    
+}
+
+// Free memory allocated by the hipMalloc(). 
+static void
+hipFree(void* ptr)
+{
+    throw std::runtime_error("Not implemented");    
+}
+
+// Register host memory so it can be accessed from the current device.
+static void
+hipHostRegister(void* hostPtr,
+                size_t sizeBytes,
+                unsigned int flags)
+{
+    throw std::runtime_error("Not implemented");   
+}
+
+// Un-register host pointer.
+static void
+hipHostUnregister(void* hostPtr)
+{
+    throw std::runtime_error("Not implemented");    
+}
+
+// Get Device pointer from Host Pointer allocated through hipHostMalloc().
+static void
+hipHostGetDevicePointer(void** devPtr,
+                        void* hstPtr,
+                        unsigned int flags)
+{
+    throw std::runtime_error("Not implemented");    
+}
+
+// Copy data from src to dst. 
+static void
+hipMemCpy(void* dst,
+          const void* src,
+          size_t sizeBytes,
+          hipMemcpyKind kind)
+{
+    throw std::runtime_error("Not implemented");    
+}
+
+} // xrt::core::hip
+
+// Allocate memory on the device.
+hipError_t
+hipMalloc(void** ptr, size_t size)
+{
+    *ptr = nullptr;
```

Comment:
Usually variables are created with minimal scope.
But at the end, it does not really seem use full. Replace by `nullptr` at the call site?

Reply:
Statement "*ptr = nullptr;" has been moved to inside of zero size check if block, since AMD HIP spec require *ptr return nullptr in this case.